pyro/ops/tensor_utils.py

```diff
@@ -61,3 +61,27 @@ def convolve(signal, kernel, mode='full'):
 
     start_idx = (padded_size - truncate) // 2
     return result[..., start_idx: start_idx + truncate]
+
+
+def parallel_scan_repeated_matmul(M, n):
```

Comment:
Can you rename this to just `repeated_matmul`? the "parallel scan" part is an implementation detail.